src/apps/unagii/ethereum/unagii.vault.token-fetcher.ts

```diff
@@ -1,126 +1,33 @@
-          multicall.wrap(uTokenContract).totalSupply(),
-          multicall.wrap(uTokenContract).decimals(),
-          multicall.wrap(uTokenContract).symbol(),
-        ]);
-        const underlyingTokenAddress =
-          underlyingTokenAddressRaw.toLowerCase() === ETH_ADDR_ALIAS
-            ? ZERO_ADDRESS
-            : underlyingTokenAddressRaw.toLowerCase();
-        const underlyingToken = await this.appToolkit.getBaseTokenPrice({ network, address: underlyingTokenAddress });
-        if (!underlyingToken) return null;
-
-        // Determine total supply
-        const underlyingPrice = underlyingToken.price || 0;
-        const supply = Number(totalSupplyRaw) / 10 ** decimals;
-        const underlyingAssets = Number(totalAssetsRaw) / 10 ** underlyingToken.decimals;
-
-        // Determine the price per share
-        const pricePerShare = underlyingAssets / supply;
-        const price = pricePerShare * underlyingToken.price;
-
-        const reserve = supply * pricePerShare;
-        const liquidity = reserve * underlyingPrice;
+export class EthereumUnagiiVaultTokenFetcher extends UnagiiVaultTokenFetcher {
+  appId = appId;
+  groupId = groupId;
+  network = network;
+  groupLabel = 'Vaults';
+  vaultManagerAddresses = [
+    '0x7f75d72886d6a8677321e5602d18948abcb4281a',
+    '0xb088d7c71ea9ebaed981c103fc3019b59950d2c9',
+    '0x8ef11c51a666c53aeeec504f120cd1435e451342',
+  ];
 
-        const tokens = [{ ...underlyingToken, reserve }];
-        const reservePercentages = tokens.map(t => reserve * (t.price / liquidity));
-        const ratio = reservePercentages.map(p => `${Math.round(p * 100)}%`).join(' / ');
-        const secondaryLabel = ratio;
-
-        const displayProps = {
-          label: symbol,
-          secondaryLabel,
-          images: [getTokenImg(underlyingToken.address)],
-          statsItems: [
-            {
-              label: 'Liquidity',
-              value: buildDollarDisplayItem(liquidity),
-            },
-            {
-              label: 'Supply',
-              value: buildNumberDisplayItem(supply),
-            },
-            {
-              label: 'Ratio',
-              value: ratio,
-            },
-          ],
-        };
-
-        const dataProps = {
-          liquidity,
-          isActive: vault.isActive,
-        };
-
-        const token: AppTokenPosition = {
-          address: tokenAddress,
-          type: ContractType.APP_TOKEN,
-          network,
-          appId,
-          groupId,
-          symbol,
-          decimals,
-          supply,
-          price,
-          pricePerShare,
-          tokens,
-          dataProps,
-          displayProps,
-        };
-
-        return token;
-      }),
-    );
-
-    return compact(vaultTokens);
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(UnagiiContractFactory) protected readonly contractFactory: UnagiiContractFactory,
+  ) {
+    super(appToolkit, contractFactory);
   }
```

Comment:
Remove constructor, the default is fine here. Only override the constructor if there are additional dependencies you need to inject.